### PR TITLE
Slackbot to message new users

### PR DESCRIPTION
**Pull request by ye-susan**
Related: https://github.com/hackforla/VRMS/issues/154
I'll have to take some time off VRMS to focus on work, so I wanted to submit my PR with what I currently have.

This PR contains commits for the Slackbot to message user accounts less than 30 days old.
Currently functionality: 
* query the db for user accounts that are <= 30 days old
* message user from markdown via Slackbot direct message

Improvements Needed:
* check if user has already attended meetings, so they won't receive the messages
* this slackbot is currently not linked to cron to be run daily
* need a way to determine that we've already messaged user so they won't receive message again

### CODE REVIEW

**Comment by alex-anakin** [comment]
@ye-susan is it possible to test it locally by a reviewer? If yes, please add steps on how to perform manual testing.

**Comment by ye-susan** [comment]
Hi @alex-anakin, 
For testing locally, the reviewer would need: 
* slack tokens for their test slack workspace
* an email to test on (email should be included user table in test db and createdDate should be less than 30 days old), the email would also need to be the same email used for the slack workspace 
(since the slack API searches for the email to find the corresponding Slack ID for the slackbot to direct message)

File: backend/routers/slack.router.js [hunk 1]
```diff
@@ -111,4 +114,63 @@ async function findProject(req, res) {
...
+async function findAndDmChat() {
+  try {
+    const potentialSlackUsers = await getUsers();
+    const actualSlackUsers = [];
+
+    //find users who are in our slack
+    for (let user of potentialSlackUsers) {
+      console.log("Email:", user.email)
+      try {
+        let result = await app.client.users.lookupByEmail({
```
Comment (nickbeaird) [inline backend/routers/slack.router.js:146]: Although I know that I was the direct cause of this implementation, please take a gander at this Google best practices on async/await for slightly refactoring this section for the better. https://developers.google.com/web/fundamentals/primers/async-functions#example_outputting_fetches_in_order

I do know that you are busy with things, so just let me know if you are too busy for that.

File: backend/routers/slack.router.js [hunk 2]
```diff
@@ -111,4 +114,63 @@ async function findProject(req, res) {
...
+        console.log("User is not on Slack");
+      }
+    }
+ 
+    const fileName = './assets/newMember.md';
+    const textData = fs.readFileSync(fileName,'utf8');
+    
+    //send Slack users msg from Slackbot
+    for (let slackUser of actualSlackUsers) {
+      const sendUserMsgObj = app.client.chat.postMessage({
```
Comment (nickbeaird) [inline backend/routers/slack.router.js:164]: See the same as the above statement.

**Review by nickbeaird — CHANGES_REQUESTED** [review]
I have not tested this manually myself, but I am throwing in a quick review on the async loops.

File: backend/routers/slack.router.js (hunk 1 above)
Comment (ye-susan) [reply backend/routers/slack.router.js:146]: Thanks for reviewing Nick! I will be a bit slammed in the coming weeks in terms of scheduling, maybe I can create an issue to improve on my PR? Either I can come back to it when I have more time -- or if this is blocking/urgent then someone else might take it up?

**Pull request closed without merging**